# Exploratory Data Analysis of Disney Datasets

## Mariwan Imbrahim

## Question(s) of Interests

In this analysis, I will be investigating a question associated with the collection of disney datasets. I am keen on finding out which type of movie had been mostly produced annually and which of them had become the most popular. It is interesting because as you may know movie has different genres and every year new movies are released. The movies sometimes become very popular and sometimes not so it is unkown. This would be very interesting to see how many movies had been produced over the years, which genre had been produced the most and which genre had been the most popular.

## Dataset Description

Disney has a long and storied history of making commercially and culturally films. Disney data consists of five datasets; `Walt Disney Animation Studios Films` provides a list of Disney animated movies and the hero/villain character names in each movie, `Movies Released by Walt Disney` contains a list of Disney movies, and their genre, gross, and MPAA ratings, `Disney animated universe characters` has a complete list of Disney characters and their voice actors, `Walt Disney Animation Studios films` provides a list of Disney animated movies and the director of each movie and `Annual gross revenues of The Walt Disney Company` is a Disney financial data chart which contains annual gross revenues by sections (includes studio entertainment, parks and resorts, etc.) from 1991-2016. These disney datasets are stored in `csv` file as below and obtained from the [data.world](https://data.world/kgarrett/disney-character-success-00-16).  
- _disney-characters.csv_
- _disney-director.csv_
- _disney-voice-actors.csv_
- _disney_revenue_1991-2016.csv_
- _disney_movies_total_gross.csv_

Based on my analysis, I will be using the **disney_movies_total_gross** table, which contains **movie_title**, **release_date**, **genre**, **MPAA_rating**, **total_gross** and **inflation_adjusred_gross**.

# Methods and Results

As the disney movie total gross provides informations about the release, movie title, genre, mpaa rating, inflation adjusted gross and total gross about Disney films and I have decided to analyze these variables that contain information on genre.

First, let's import the table and do some basic visualizations.

In [1]:
# Let's import the required libaraies for this analysis
import pandas as pd
import altair as alt

# Import the dataset
disney = pd.read_csv('data/disney_movies_total_gross.csv')

Let's see what the table looks like.

In [2]:
disney.head()

movie_title  release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs  Dec 21, 1937    Musical           G   
1                        Pinocchio   Feb 9, 1940  Adventure           G   
2                         Fantasia  Nov 13, 1940    Musical           G   
3                Song of the South  Nov 12, 1946  Adventure           G   
4                       Cinderella  Feb 15, 1950      Drama           G   

    total_gross inflation_adjusted_gross  
0  $184,925,485           $5,228,953,251  
1   $84,300,000           $2,188,229,052  
2   $83,320,000           $2,187,090,808  
3   $65,000,000           $1,078,510,579  
4   $85,000,000             $920,608,730

First of all, it’s so much important to have a know of the data. So, let’s get some information about the table.

In [3]:
disney.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 579 entries, 0 to 578
Data columns (total 6 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   movie_title               579 non-null    object
 1   release_date              579 non-null    object
 2   genre                     562 non-null    object
 3   MPAA_rating               523 non-null    object
 4   total_gross               579 non-null    object
 5   inflation_adjusted_gross  579 non-null    object
dtypes: object(6)
memory usage: 27.3+ KB


In [4]:
disney.describe(include = 'all')

movie_title  release_date   genre MPAA_rating total_gross  \
count               579           579     562         523         579   
unique              573           553      12           5         576   
top     The Jungle Book  Dec 25, 1997  Comedy          PG          $0   
freq                  3             3     182         187           4   

       inflation_adjusted_gross  
count                       579  
unique                      576  
top                          $0  
freq                          4

In the above outputs, the disney table has 579 rows and 6 columns. The movie title has 12 types of genre, 5 types of MPAA rating, the date they were released, total gross and inflation adjusted gross. As It is seen that genre has a few missing values and MPAA rating contains 56 missing values. Note that MPAA rating is not a related variable to my analysis so I prefer leaving the data the way is it. Moreover, the release date, total gross and inflation adjusted gross do not have the right data types. 

Let's drop these missing values in genre variable and then check the table again.  

In [5]:
# This drops missing values only in genre
disney_drop = disney.dropna(subset = ['genre'])
disney_drop.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 562 entries, 0 to 578
Data columns (total 6 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   movie_title               562 non-null    object
 1   release_date              562 non-null    object
 2   genre                     562 non-null    object
 3   MPAA_rating               513 non-null    object
 4   total_gross               562 non-null    object
 5   inflation_adjusted_gross  562 non-null    object
dtypes: object(6)
memory usage: 30.7+ KB


Now it is time to convert release date to datetime, inflation adjusted gross and total gross to integer. First, removing the dollar sign and comma. 

In [6]:
# Import the created function to convert release date object type to datetime and
# remove $ sign and comma in total gross and inflation adjusted gross and convert them to interger.
from project_function import data_trans
disney_trans = data_trans(disney_drop, 'release_date', 'total_gross', 'inflation_adjusted_gross')
disney_trans.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 562 entries, 0 to 578
Data columns (total 6 columns):
 #   Column                    Non-Null Count  Dtype         
---  ------                    --------------  -----         
 0   movie_title               562 non-null    object        
 1   release_date              562 non-null    datetime64[ns]
 2   genre                     562 non-null    object        
 3   MPAA_rating               513 non-null    object        
 4   total_gross               562 non-null    int64         
 5   inflation_adjusted_gross  562 non-null    int64         
dtypes: datetime64[ns](1), int64(2), object(3)
memory usage: 30.7+ KB


Perfect! let's create a frequency table for genre and visualize it. 

In [7]:
# This creates frequency table of genre, resets the index and renames the columns so we can plot it. 
ferq_genre = (disney_trans['genre'].value_counts(ascending = True).
              reset_index().
              rename(columns = {'index':'genre', 'genre':'count'}))
ferq_genre

genre  count
0   Concert/Performance      2
1          Black Comedy      3
2                Horror      6
3               Western      7
4           Documentary     16
5               Musical     16
6       Romantic Comedy     23
7     Thriller/Suspense     24
8                Action     40
9                 Drama    114
10            Adventure    129
11               Comedy    182

Genre frquency table is in the proper format so we can generate a bar plot to visualize it.

In [8]:
# Use altair to generate bar plot
genre_bar_plot = alt.Chart(ferq_genre, width = 500, height = 300).mark_bar().encode(
    x = alt.X('genre:N', sort = 'y'),
    y = alt.Y('count:Q')
).properties(title = 'Total Number of Each Genre')

genre_bar_plot

alt.Chart(...)

From the above bar plot, It seems that comedy movies had been produced the most since 1937 and adventure movie is the second and third is drama. 

Let's extract year from release date named year

In [9]:
# Create a new column name year extracted from release date.
disney_year = disney_trans.assign(year = disney_trans['release_date'].dt.year)
disney_year.head()

movie_title release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                        Pinocchio   1940-02-09  Adventure           G   
2                         Fantasia   1940-11-13    Musical           G   
3                Song of the South   1946-11-12  Adventure           G   
4                       Cinderella   1950-02-15      Drama           G   

   total_gross  inflation_adjusted_gross  year  
0    184925485                5228953251  1937  
1     84300000                2188229052  1940  
2     83320000                2187090808  1940  
3     65000000                1078510579  1946  
4     85000000                 920608730  1950

Then let's group disney_year by year and count all types of movies anuually. 

In [10]:
# This groups the data by year and compute the total number of movies for each year.
# Reset the table and rename the column genre to count.
year_group = (disney_year.groupby(by = 'year').
              agg(func = {'genre':'count'}).
              reset_index().
              rename(columns = {'genre':'count'}))

year_group.head(10)

year  count
0  1937      1
1  1940      2
2  1946      1
3  1950      1
4  1954      1
5  1955      1
6  1959      1
7  1961      3
8  1962      1
9  1963      1

The year_group table contains total movies for each year and shows only the first 10 rows.

Let's generate a line plot for above table.

In [11]:
# Use altair to make line plot.
year_line_plot = alt.Chart(year_group, width = 500, height = 300).mark_line().encode(
    x = alt.X('year:O'),
    y = alt.Y('count:Q')
).properties(title = 'Total Number of Movies Produced Annually')
year_line_plot

alt.Chart(...)

From the above line plot, from 1937 to 1975, there are not many movies released. After 1975, it is slightly increased until 1983. As it seen there is a small drop from 1983 to 1984, but then it rises rapidly til 1994. It drops dramatically after 1994 then it flactuates. The plot shows that the majority of movies were released from 1984 til the end and it will continue to grow.

Let's group the data by genre and year and then count each genre. That is very interesting to see how many movies from each genre were produced over the years.  

In [12]:
# Group disney_year by genre and year.
# Count each genre movie in each year, then reset the index and rename the column to count.
disney_genre_year = (disney_year.groupby(by = ['genre', 'year']).
                     size().
                     reset_index().
                     rename(columns = {0:'count'}))
disney_genre_year

genre  year  count
0     Action  1981      1
1     Action  1982      1
2     Action  1988      2
3     Action  1990      2
4     Action  1991      2
..       ...   ...    ...
213  Western  1995      1
214  Western  2000      1
215  Western  2003      1
216  Western  2004      2
217  Western  2013      1

[218 rows x 3 columns]

The frequency table of genre and year is in the proper format so let's visualize it.

In [13]:
# Use altair to generate line plots.
genre_year_plot = alt.Chart(disney_genre_year, width = 200, height = 200).mark_bar().encode(
    x = alt.X('year:O'),
    y = alt.Y('count:Q'),
    color = alt.Color('genre:N'),
    facet = alt.Facet('genre:N', columns = 3)
).resolve_axis(
    x='independent',
    y='independent',
).properties(title = 'Produced Each Genre Over Years')

genre_year_plot

alt.Chart(...)

In the above bar plots, it seems that __Comedy__ gets produced and released after 1960. Between the years 1986 and 2008, a lot of comedy movies were released. __Adventure__ movies were produced and released in 1930s, but not too many until 1986. After that the number of adventrue movies increases and descreases slightly over the years. A few __Darama__ movies were released between 1937 and 1980, but then the number of drama movies goes up specifically between 1993 and 2000. __Action__ has no released movies until 1980 and after that there are several action movies release during the years.  

Let's take a look at total gross variable and get some information.

In [14]:
disney_year.describe()['total_gross']

count    5.620000e+02
mean     6.644891e+07
std      9.384227e+07
min      0.000000e+00
25%      1.417648e+07
50%      3.201474e+07
75%      7.792766e+07
max      9.366622e+08
Name: total_gross, dtype: float64

The minimum value of total gross is zero. Let's sort the data ascendingly by total gross and see how many zero values it has. 

In [15]:
# Sort the data by total gross ascendly
disney_gross = disney_year.sort_values(by = 'total_gross')
disney_gross.head(10)

movie_title release_date        genre  \
27                                      Amy   1981-03-20        Drama   
29                                Condorman   1981-08-07       Action   
511                                Zokkomon   2011-04-22    Adventure   
487                       Walt and El Grupo   2009-09-10  Documentary   
280  An Alan Smithee Film: Burn Hollywood …   1998-02-27       Comedy   
502        Gedo Senki (Tales from Earthsea)   2010-08-13    Adventure   
185                                It's Pat   1994-08-26       Comedy   
495                  Waking Sleeping Beauty   2010-03-26  Documentary   
311                  Breakfast of Champions   1999-09-17       Comedy   
470            Goal! 2: Living the Dream...   2008-08-29        Drama   

    MPAA_rating  total_gross  inflation_adjusted_gross  year  
27          NaN            0                         0  1981  
29          NaN            0                         0  1981  
511          PG         2815                      2984  2011  
487          PG        20521                     23064  2009  
280           R        45779                     82277  1998  
502       PG-13        48658                     51988  2010  
185         NaN        60822                    125666  1994  
495          PG        80741                     86264  2010  
311           R       178287                    295851  1999  
470       PG-13       225067                    264247  2008

There are only two missing values in total gross. To fix these data, let's replace these null values by median value of total gross.

In [16]:
# This computes median value of total gross
gross_median = disney_year['total_gross'].median()

# This fills the null values by the median value of total gross
disney_year.loc[disney_year['total_gross'] == 0, 'total_gross'] = gross_median

disney_year.describe()['total_gross']

count    5.620000e+02
mean     6.656284e+07
std      9.378084e+07
min      2.815000e+03
25%      1.472850e+07
50%      3.201632e+07
75%      7.792766e+07
max      9.366622e+08
Name: total_gross, dtype: float64

Now the total gross looks great.

Let's generate a histogram plot for total gross and see if it is normally distributed.

In [17]:
# Use altair to make a histogram plot.
gross_hist = alt.Chart(disney_year, width = 500, height = 300).mark_bar().encode(
    x = alt.X('total_gross:Q', bin = alt.Bin(maxbins = 50)),
    y = alt.Y('count()')
).properties(title = 'Distribution of Total Gross')
gross_hist

alt.Chart(...)

In the above histogram, there appears that there are some outliers in the total gross and these outliers will not be an issue in this analysis so I will not make a tranformation and leave the way is it. 

Let's calculate the sum of total gross for each genre.

In [18]:
# Group the data by genre, sum total gross and reset the index.
# Sort total gross descendingly.
genre_gross = (disney_year.groupby(by = 'genre').
              agg(func = {'total_gross':'sum'}).
              reset_index().
              sort_values(by = 'total_gross', ascending = False))
genre_gross

genre   total_gross
1             Adventure  1.638907e+10
3                Comedy  8.119620e+09
0                Action  4.216578e+09
6                 Drama  4.138988e+09
10    Thriller/Suspense  1.406807e+09
8               Musical  1.157284e+09
9       Romantic Comedy  1.152207e+09
11              Western  3.590115e+08
5           Documentary  1.806856e+08
4   Concert/Performance  1.034565e+08
2          Black Comedy  9.754321e+07
7                Horror  8.706887e+07

In above table, It seems __Adventrue__ has the highest total gross, which means it was the most popular movie over the years. The second of most popular movie was __Comedy__, the third was __Action__ movies and __Drama__ has very close total gross to action movie. 

# Discussions

In this work, I analyzed the disney dataset **disney_movies_total_gross** and tried to find out how many movies had been produced, what genres had been the most released over the years and which of genre was the most popular. In this analysis, the missing values of genre was removed and the few null values of total gross was filled by the mediam of total gross. I found that there is an increasing trend of producing movies from 1937 to 1994, but then it goes down so in general it could be said there is a significant increasing number movies.

In addition, the majorit of producing movies are **Comedy**, **Adventure**, **Drama** and **Action** in order. Comedy has the hgihest number of movies among others and adventure and drame have a close numbers of their movies. In general, they were mostly released after 1980 and it can be seen that disney developed producing movies after 1980. 

It is very surprising to see __Adventure__ has the highest total gross, by that means adventure is the most popular movie. __Comedy__ has the second high total gross and is the second popular movie, even though Comedy has way more movies than adventure. **Action** has a higher total gross then **Drama**, which is another surprise. As we know drama has almost triple number of moives compared to action, but action movie is more popular than dram. The result is out of my expectations! I expected that Comedy movies would be the most popular and then adventure, drama and action movies.

I recommend a few more questions for further analysis on this dataset are what is the highest MPAA rating over the years and which one is the most popular. These can be done together and for this recommendation, the inflation adjusted gross might be made use of. This will be more interesting because movies are produced and released not for all ages and sometimes there is a limited age to watch the mvoie. By this analysis, it might know the movies should be produced more for kind of MPAA rating.

# References

Not all the work in this notebook is original. Some parts of dataset description were borred from the data world website.